3rdparty/everest/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+list (APPEND everest_src)
+list (APPEND everest_inc)
+list (APPEND everest_def)
+
+execute_process(COMMAND ${PERL_EXECUTABLE} ../scripts/config.pl -f ../include/mbedtls/config.h get MBEDTLS_ECDH_VARIANT_EVEREST_ENABLED RESULT_VARIABLE result)
+
+if(${result} EQUAL 0)
+  set(everest_src
+    ${CMAKE_CURRENT_SOURCE_DIR}/library/everest.c
+    ${CMAKE_CURRENT_SOURCE_DIR}/library/x25519.c
+    ${CMAKE_CURRENT_SOURCE_DIR}/library/kremlib/FStar_UInt64_FStar_UInt32_FStar_UInt16_FStar_UInt8.c
+  )
+
+  if (${CMAKE_LIBRARY_ARCHITECTURE} STREQUAL "x86_64-linux-gnu")
```

Comment:
Fix for the error in 8434e9bbcbcb7e1e142e899a4edbbf6fb2804434. It looks like old CMake's don't provide `CMAKE_LIBRARY_ARCHITECTURE`; it works fine on my machine. After adding the missing quotes, the check will fail when using old CMake's, i.e. the legacy version of the code will be used (ok but slower).

Reply:
I was using cmake 3.5.1 on Ubuntu 16.04. [It should be supported](https://cmake.org/cmake/help/v3.5/variable/CMAKE_LIBRARY_ARCHITECTURE.html) “if detected”. Do you have any idea what “if detected” means?

Reply:
I have no idea! Is there a cmake expert we could ask about this? BTW, that's not an "if", it "as detected", implying that `CMAKE_C_LIBRARY_ARCHITECTURE` should work for C. Perhaps it would be sufficient to change `CMAKE_LIBRARY_ARCHITECTURE` to `CMAKE_C_LIBRARY_ARCHITECTURE` (with/without `_C_`)?

Reply:
Unfortunately I don't know any cmake expert. The first sentence in the manual does say “if detected”. In any case cross-compilation to x86 is relatively rare so we can live with a less-optimized version in this case for now.